src/aof.c

```diff
@@ -265,7 +417,45 @@ void flushAppendOnlyFile(int force) {
      * or alike */
 
     latencyStartMonitor(latency);
-    nwritten = write(server.aof_fd,server.aof_buf,sdslen(server.aof_buf));
+#ifdef USE_NVML
+    if (server.aof_use_nvml == 1) {
+        size_t len = sdslen(server.aof_buf);
+        size_t newlen = server.aof_current_size + len;
+
+        /* Check if there's required space, if not try to rewrite first */
+        if (newlen > server.aof_nvml_log_size)
+        {
+            /* if there's not enough space try to compact AOF file */
+            redisLog(REDIS_ERR, "Not enough space to write AOF file");
+            redisLog(REDIS_NOTICE, "Starting background rewrite of AOF file");
+            if (rewriteAppendOnlyFileBackground() != REDIS_OK) {
+                redisLog(REDIS_ERR, "Starting background rewrite of AOF file failed!");
+            }
+            nwritten = -1;
+            errno = ENOSPC;
+        } else {
+            if (!pmemlog_append(server.aof_plp, server.aof_buf, len)) {
+                nwritten = len;
+
+                /* if reached threshold for AOF rewrite it */
+                if (server.aof_rewrite_perc > 0 &&
+                        server.aof_current_size > (server.aof_nvml_log_size *
```

Comment:
Generally we could achieve this without any changes: if we set `aof_rewrite_perc` to 1 and `aof_rewrite_min_size` to desired value.